config.yaml

```diff
@@ -2,6 +2,10 @@
 # See LICENSE file for licensing details.
 
 options:
+  external_hostname:
+      type: string
+      description: "External hostname this Synapse instance responds to. Defaults to application name."
+      default: ""
```

Comment:
The ingress relation provided by Traefik does not use this config, only nginx-route will.

Reply:
The nginx-ingress-integrator charm does provide a `service-hostname` (and `additional-hostnames`) config option. However, the nginx-route integration doesn't specify that this is provided by the nginx-ingress-integrator charm over the relation and I'm not sure if the charm does this in code. We could update the nginx-route charm so that the provider can optionally provide this over the relation though, and this would mean the requiring charm didn't need to specify `service-hostname`. Or we could just make `service-hostname` optional over the relation in the case that the provider sets it.

Reply:
We I really don't think `external_hostname` is needed by the charm, as it was added with `nginx-route` relation. So I think we can just get rid of it as suggested no ?

Reply:
Leaving it to "None" should do the job

Reply:
Changed, thanks!

Reply:
I did check if "None" was supported from requirer perspective, but didn't check from provider, and indeed the field is required from provider perspective
I'm wondering if we shouldn't make is optional if the provider can use the value configured on its side, but not for this PR